internal/session/serve.go

```diff
@@ -87,24 +88,43 @@ func (s *Session) serve() {
 		case "sethealthy":
 			log.Logger.Infow("sethealthy received", "components", payload.Components)
 			for _, componentName := range payload.Components {
-				if componentName != "accelerator-nvidia-error-xid" {
-					continue
-				}
-				rawComponent, err := components.GetComponent("accelerator-nvidia-error-xid")
-				if err != nil {
-					log.Logger.Errorw("failed to get component", "error", err)
-					continue
-				}
-				if watchable, ok := rawComponent.(*metrics.WatchableComponentStruct); ok {
-					if component, ok := watchable.Component.(*xid.XIDComponent); ok {
-						if err = component.SetHealthy(); err != nil {
-							log.Logger.Errorw("failed to set xid healthy", "error", err)
+				switch componentName {
+				case "accelerator-nvidia-error-xid":
```

Comment:
I'm curious why you only handle these two components. It seems that the logic is general

Reply:
Use 

```go
nvidia_component_error_xid_id "github.com/leptonai/gpud/components/accelerator/nvidia/error/xid/id"
nvidia_component_error_xid_id.Name
```

?

Reply:
yes updated package name @gyuho 
yes SetHealthy logic is generic, finally we should make SetHealthy a generic interface to all Components, currently this only applies to xid/sxid so didn't add the interface thought can do it later.. @eahydra

Reply:
Let's create an issue for it?